### PR TITLE
[Draft] Add GENERATE_ARRAY function

### PR DESCRIPTION
**Pull request by dttung2905**
TODO list:
- [x] implement `transformToStringValuesMV` method
- [x] testing
- [x] Register the batch function

Instructions:
1. The PR has to be tagged with at least one of the following labels (*):
   1. `feature`
   2. `bugfix`
   3. `performance`
   4. `ui`
   5. `backward-incompat`
   6. `release-notes` (**)
2. Remove these instructions before publishing the PR.
 
(*) Other labels to consider:
- `testing`
- `dependencies`
- `docker`
- `kubernetes`
- `observability`
- `security`
- `code-style`
- `extension-point`
- `refactor`
- `cleanup`

(**) Use `release-notes` label for scenarios like:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+  }
+
+  @Override
+  public byte[][] transformToBytesValuesSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
+  }
+
+  @Override
+  public int[][] transformToIntValuesMV(ValueBlock valueBlock) {
+    return new int[0][];
```
Comment (xiangfu0) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java:190]: Here we should convert array literal and cache it.
Check how `ArrayLiteralTransformFunction` did here.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+  @Nullable
+  @Override
+  public Dictionary getDictionary() {
+    return null;
+  }
+
+  @Override
+  public int[] transformToDictIdsSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
```
Comment (xiangfu0) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java:145]: throw new UnsupportedOperationException();

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java (hunk 1 above)
Comment (dttung2905) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java:190]: Yes. Its on my to do list too. No worries :smile:

File: pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java (hunk 2 above)
Comment (dttung2905) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java:145]: will fix it and the relevant functions as well. I was speeding through this yesterday sorry

**Comment by dttung2905** [comment]
Hi @xiangfu0 sorry I was really busy with works ( pre-Christmas period :sweat_smile: ). I just fixed the unit test for the failed test `testCalciteFunctionMapAllRegistered` by whitelisting it under `IGNORED_FUNCTION_NAMES`. The next time CI is run, the tests should pass.

Is there anything I'm mising before the review ? Do I need to 
- add the test for scalar function changes in `pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java`. I look up the other scalar function but can't find any test for those. cmiiw :thinking: 
- remove the changes in `pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java` as we have implemented the batch function already?

**Pull request closed without merging**